lib/rules/one-var.js

```diff
@@ -255,7 +255,9 @@ module.exports = {
 
             if (currentOptions.uninitialized === MODE_ALWAYS && currentOptions.initialized === MODE_ALWAYS) {
                 if (currentScope.uninitialized || currentScope.initialized) {
-                    return false;
+                    if (!hasRequires) {
```

Comment:
Should this also be using `separateRequires`? I can't do a deep dive at the moment, but this caught my eye.

Reply:
instead of separate requires?

Reply:
@kaicataldo could you expand on your question, dont think I understand what youre asking.

Reply:
Sorry, got a bit busy and lost track of this. My question was about these checks always checking if there are `require`s but not checking if the `separateRequires` config option was set or not. Hopefully I'll be able to look at this a little closer later tonight.

Reply:
@kaicataldo I see.  Good point.  The false being returned in my contributions allow this conditional to be fulfilled. https://github.com/eslint/eslint/blob/56a677c45f6555096dc50b19806f243c17d994ef/lib/rules/one-var.js#L441.  Though thats a good point, for this ticket i could add `options.separateRequires && !hasRequires`.

WDYT?